app/controllers/teams_controller.rb

```diff
@@ -39,6 +39,17 @@ def update
     change_description(@team, :team)
   end
 
+  # GET /teams/1/typeahead/%QUERY
+  def typeahead
+    authorize @team
+    @query = params[:query]
+    matches = User.search_from_query(@team.member_ids,
+                                     "#{@query}%").map { |user| { username: user } }
```

Comment:
IMHO: its a code smell to fetch and build objects where only a single attribute is needed.